v2/pkg/config/config.go

```diff
@@ -150,6 +151,29 @@ func ProvideInfrastructureAwareConfig(
 			return nil, errors.Wrap(err, "failed to parse config")
 		}
 
+		// Use v4.6 images on Openshift 4.6 instead of default v4.5 images
+		// But otherwise respect an override
+		if inf.OpenshiftParsedVersion().GTE(utils.ParsedVersion460) {
+			if cfg.RelatedImages.Prometheus == "registry.redhat.io/openshift4/ose-prometheus:v4.5" {
```

Comment:
This is a little wordy but I think it's fine for now. Sometimes better to be explicit.